runtime/vm/createramclass.cpp

```diff
@@ -3675,7 +3672,11 @@ internalCreateRAMClassFromROMClassImpl(J9VMThread *vmThread, J9ClassLoader *clas
 						J9ARRAYCLASS_SET_STRIDE(ramClass, J9_VALUETYPE_FLATTENED_SIZE(elementClass));
 					}
 				} else {
-					if (J9_IS_J9CLASS_ALLOW_DEFAULT_VALUE(elementClass)) {
+					if (J9_IS_J9CLASS_ALLOW_DEFAULT_VALUE(elementClass)
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+							&& J9_ARE_ALL_BITS_SET(options, J9_FINDCLASS_FLAG_CLASS_OPTION_NULL_RESTRICTED_ARRAY)
+#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
+					   ) {
```

Comment:
Please correct the indentation of line 3679; it should be indented the same as line 3675.